### PR TITLE
add HiHFFilterPF, remove XTh6 in hffilter (tower based) which is not …

### PR DESCRIPTION
**Pull request by boundino**
#### PR description:

- Add HiHFFilterPF, which gives pphfCoincFilterPFXThY up to 5Th10 using particle flow
- Remove pphfCoincFilterXTh6 in HiHFFilter (towerMaker based) which is not meaningful [ref](https://github.com/cms-sw/cmssw/blob/master/DataFormats/HeavyIonEvent/interface/HFFilterInfo.h)
- Move list of XThY to _cfi from forest_.py
- Use 2024 PbPb prompt reco file as test input

### CODE REVIEW

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  numMinHFTowersM = 0;
+
+  edm::Handle<pat::PackedCandidateCollection> pfCandidates;
+  iEvent.getByToken(pfCandidateTag_, pfCandidates);
+
+  for (const auto& pfcand : *pfCandidates) {
+
+    /* dummy reco::PFCandidate used to convert pdgId */
+    auto id = converter_.translatePdgIdToType(pfcand.pdgId());
+    if (!(id == 6 || id == 7)) continue;
```
Comment (stahlleiton) [inline HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: Could be simplified by changing to:

if (pfcand.pdgId() != 1 && pfcand.pdgId() != 2)

then we can remove the "converter_" from the code

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+public:
+  explicit HiHFFilterPF(const edm::ParameterSet&);
+  ~HiHFFilterPF() override;
+
+private:
+  void beginJob() override;
+  bool filter(edm::Event&, const edm::EventSetup&) override;
+
+  // edm::EDGetTokenT<reco::HFFilterInfo> HFfilters_;
+  edm::EDGetTokenT<pat::PackedCandidateCollection> pfCandidateTag_;
```
Comment (stahlleiton) [inline HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: Set class member (lines 29 to 32) to const (e.g. const double threshold_, ....)

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+HiHFFilterPF::~HiHFFilterPF() {}
+
+bool HiHFFilterPF::filter(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  bool accepted = false;
+
+  numMinHFTowersP = 0;
+  numMinHFTowersM = 0;
+
+  edm::Handle<pat::PackedCandidateCollection> pfCandidates;
+  iEvent.getByToken(pfCandidateTag_, pfCandidates);
```
Comment (stahlleiton) [inline HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: Can be simplified by doing:

const auto& pfCandidates = iEvent.get(pfCandidateTag_);

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    /* dummy reco::PFCandidate used to convert pdgId */
+    auto id = converter_.translatePdgIdToType(pfcand.pdgId());
+    if (!(id == 6 || id == 7)) continue;
+        
+    // float pt = pfcand.pt();
+    float eta = pfcand.eta();
+    if (std::abs(eta) > 6 || std::abs(eta) < 3) {
+      continue;
+    }
```
Comment (stahlleiton) [inline HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: const auto abseta = std::abs(pfcand.eta());
if (abseta > 6 || abseta < 3)
  continue;

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    if (pfcand.energy() >= threshold_) {
+      if (eta > 0) { numMinHFTowersP++; }
+      else { numMinHFTowersM++; }
+    }
+    
+  } // for (const auto& pfcand : *pfCandidates)
+  if (std::min(numMinHFTowersP, numMinHFTowersM) >= minnumtowers_)
```
Comment (stahlleiton) [inline HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: if (numMinHFTowersP >= minnumtowers_ && numMinHFTowersM >= minnumtowers_)

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    accepted = true;
+
+  return accepted;
+}
+
+void HiHFFilterPF::beginJob() {
```
Comment (stahlleiton) [inline HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: Could remove if converter not used

**Comment by stahlleiton** [comment]
Added comments for review

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc (hunk 1 above)
Comment (boundino) [reply HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: done

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc (hunk 3 above)
Comment (boundino) [reply HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: done

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc (hunk 4 above)
Comment (boundino) [reply HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: done

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc (hunk 5 above)
Comment (boundino) [reply HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: done

File: HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc (hunk 6 above)
Comment (boundino) [reply HeavyIonsAnalysis/EventAnalysis/plugins/HiHFFilterPF.cc]: done

**Pull request merged**